src/metrics/kms-service.ts

```diff
@@ -1,36 +1,21 @@
-// This only imports the type of KMS, not the class. As we don't do an instantiation typescript will compile this away.
-import { KMS } from "aws-sdk";
-
-import { logError } from "../utils";
-
 // In order to avoid the layer adding the 40mb aws-sdk to a deployment, (which is always available
 // in the lambda environment anyway), we use require to import the sdk, and return an error if someone
 // tries to decrypt a value.
 export class KMSService {
-  private kms?: KMS;
 
-  constructor() {
+  public async decrypt(value: string): Promise<string> {
     try {
-      const kmsType = require("aws-sdk").KMS;
-      this.kms = new kmsType() as KMS;
+      const kmsType = require("aws-sdk/clients/kms");
+      const kms = new kmsType();
+      const buffer = Buffer.from(value, "base64");
+      const encryptionContext = { LambdaFunctionName: process.env.AWS_LAMBDA_FUNCTION_NAME ?? "" };
+      const result = await kms.decrypt({ CiphertextBlob: buffer, EncryptionContext: encryptionContext }).promise();
+      if (result.Plaintext === undefined) {
+        throw Error("Couldn't decrypt value");
+      }
+      return result.Plaintext.toString("ascii");
     } catch (err) {
-      logError("optional dependency aws-sdk not installed. KMS key decryption will not work", err);
-    }
-  }
-
-  public async decrypt(value: string): Promise<string> {
-    if (this.kms === undefined) {
-      // This error should only occur if the user is running this code outside of the lambda environment,
-      // (say a local development environment).
       throw Error("optional dependency aws-sdk not installed. KMS key decryption will not work");
```

Comment:
I think we should log original error before throwing again.

Reply:
✅